cbor-schema/plt.cddl

```diff
@@ -163,3 +163,34 @@ token-initialization-parameters = {
     ; Whether the token is burnable
     ? "burnable": bool .default false
 }
+
+; Token state
+
+; The Token Module state represents global state information that is maintained by the Token Module,
+; and is returned as part of a `GetTokenInfo` query. It does not include state that is managed by
+; the Token Kernel, such as the token identifier, global supply and governance account. It also
+; does not (typically) include account-specific state, which is returned as part of
+; `GetAccountInfo` instead.
+;
+; The "name" and "metadata" fields are required. Other fields are optional, and can be omitted if
+; the module implementation does not support them. The structure supports additional fields for
+; future extensibility. Non-standard fields (i.e. any fields that are not defined by a standard,
+; and are specific to the module implementation) may be included, and their tags should be
+; prefixed with an underscore ("_") to distinguish them as such.
```

Comment:
I don't think I understand the distinction made between "standard" and "specific to module implementation" here. Are we not supposed to be concrete in the CBOR schema, i.e. _not_ allow flexibility? I feel like the "flexibility" part is already covered by choosing to CBOR encode the parts of the PLT data that is variable between token modules.

Maybe I'm misunderstanding something here, but I would imagine 1-1 relationship between token modules and CBOR schemas.

Reply:
After discussion in office, the conclusion is that the **cddl defines a plt standard** which can have **different implementations of the standard, i.e. plt module versions**.

In that case, it makes sense to me that there is support for arbitrary fields for different implementations.

Reply:
My intention is that the schema defines an interface for a PLT module that can have multiple implementations. The schema can allow for implementation defined behaviour (adding non-standard fields), but is also intended to still allow for an extension to the schema in a subsequent standard.

Reply:
I'm also confused about this. If we have _standards_, should those be defined in CBOR at all? Or can we update the standard defined in CBOR without a protocol update?

Reply:
The idea is that eventually we will have WASM-implemented token modules, and we can deploy and update new ones without a protocol update. These can have additional functionality that is specified by a subsequent standard.